examples/mixer_demo.rs

```diff
@@ -11,10 +11,12 @@ fn main() {
 
     let args: Vec<_> = env::args().collect();
 
-    if args.len() < 3 {
-        println!("Usage: ./demo music.[mp3|wav|ogg] sound-effect.[mp3|wav|ogg]")
-    } else {
+    if args.len() == 2 {
+        demo(Path::new(&args[1]), Path::new(""));
+    } else if args.len() == 3 {
```

Comment:
This is a pretty un-rusty way to do this, in Rust we have nice things like options instead of empty strings, so let's use these instead! Make `demo` have this signature: `demo(music_path: &Path, sound_path: Option<&Path>)`, and just pass args.get(2) to actually have an `Option<&Path>`. That effectively means merging the branches `args.len() == 2` and `args.len() == 3` as one here.

If the option is None, don't output anything. If it's Some(path) but the path is wrong, output an error, and if it's a correct sound file, just output the same thing.